### ISSUE TITLE
Part 7. Brief Introduction of Deep Learning

### ISSUE THREAD

**Issue by Calemsy** [XuSShuai/Machine-Learning-2017-Fall#7]
## Three Steps for Deep Learning

 - Step 1: define a set of functions(Neural Network)
 - Step 2: goodness of functions
 - Step 3: pick the best function

## Step 1: Neural Network

神经元不同的连接方式会导致不同的网络结构，即不同的function set。当固定了神经网络的架构，那么就定义了function set。
![image](https://user-images.githubusercontent.com/29543183/49136512-c8522900-f324-11e8-9b4d-7e52ceb7ddca.png)

 - Fully Connected Network
相邻两层之间的神经元都有连边的神经网络成为全连接神经网络。
![image](https://user-images.githubusercontent.com/29543183/49136642-31d23780-f325-11e8-9257-60cd85dedd1d.png)

 - Matrix Operation
![image](https://user-images.githubusercontent.com/29543183/49136885-f1bf8480-f325-11e8-830f-ed29306993ca.png)

 - Neural Network
![image](https://user-images.githubusercontent.com/29543183/49137004-36e3b680-f326-11e8-8f43-b48c293a6d83.png)

Deep learning将问题从如何抽取feature转换为如何设计好的神经网络结构。

## Step 2: Loss for Training Data

 - loss for an Example
![image](https://user-images.githubusercontent.com/29543183/49137613-ecfbd000-f327-11e8-8484-21aedd83cd23.png)

 - loss for Total
![image](https://user-images.githubusercontent.com/29543183/49137653-0e5cbc00-f328-11e8-8bfe-62b8717419a9.png)

## Step 3: Gradient Descent

![image](https://user-images.githubusercontent.com/29543183/49137711-3815e300-f328-11e8-9973-36a3f3fa174e.png)
